video/out/gpu/d3d11_helpers.c

```diff
@@ -509,12 +771,57 @@ static bool configure_created_swapchain(struct mp_log *log,
                       requested_format :
                       (probed_format != DXGI_FORMAT_UNKNOWN ?
                        probed_format : DXGI_FORMAT_R8G8B8A8_UNORM);
+    selected_colorspace = requested_csp != -1 ?
+                          requested_csp : probed_colorspace;
+    format_name = d3d11_get_format_name(selected_format);
+    csp_name    = d3d11_get_csp_name(selected_colorspace);
+    mp_csp      = d3d11_get_mp_csp(selected_colorspace);
 
-    mp_verbose(log, "Selected swapchain format %d, attempting "
+    mp_verbose(log, "Selected swapchain format %s (%d), attempting "
                     "to utilize it.\n",
-               selected_format);
+               format_name, selected_format);
+
+    if (!update_swapchain_format(log, swapchain, selected_format)) {
+        return false;
+    }
 
-    return update_swapchain_format(log, swapchain, selected_format);
+    if (!IsWindows10OrGreater()) {
+        // On older than Windows 10, query_output_format_and_colorspace
+        // will not change probed_colorspace, and even if a user sets
+        // a colorspace it will not get applied. Thus warn user in case a
+        // value was specifically set and finish.
+        if (requested_csp != -1) {
+            mp_warn(log, "User selected a D3D11 color space %s (%d), "
+                         "but configuration of color spaces is only supported"
+                         "from Windows 10! The default configuration has been "
+                         "left as-is.\n",
+                    csp_name, selected_colorspace);
+        }
+
+        return true;
+    }
+
+    if (mp_csp.gamma == MP_CSP_TRC_AUTO) {
+        mp_warn(log, "Color space %s (%d) does not have an mpv color space "
+                     "mapping! Overriding to standard sRGB!\n",
+                csp_name, selected_colorspace);
+        selected_colorspace = DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
```

Comment:
This is another place where sRGB is used to refer to DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709. Also, I wonder if it should just use TRC_AUTO/PRIM_AUTO when the colourspace mapping is unknown.

Reply:
Ditto regarding the G22 mapping stuff.